geode-core/src/distributedTest/java/org/apache/geode/internal/cache/partitioned/fixed/FixedPartitioningTestBase.java

```diff
@@ -1191,29 +1188,12 @@ public static void setPRObserverBeforeCalculateStartingBucketId() {
     origObserver =
         PartitionedRegionObserverHolder.setInstance(new PartitionedRegionObserverAdapter() {
           public void beforeCalculatingStartingBucketId() {
-            WaitCriterion wc = new WaitCriterion() {
-              String excuse;
-
-              public boolean done() {
-                Region prRoot = PartitionedRegionHelper.getPRRoot(cache);
-                PartitionRegionConfig regionConfig = (PartitionRegionConfig) prRoot.get("#Quarter");
-                if (regionConfig == null) {
-                  return false;
-                } else {
-                  if (!regionConfig.isFirstDataStoreCreated()) {
-                    return true;
-                  } else {
-                    return false;
-                  }
-                }
-              }
-
-              public String description() {
-                return excuse;
-              }
-            };
-            Wait.waitForCriterion(wc, 20000, 500, false);
-            LogWriterUtils.getLogWriter().info("end of beforeCalculatingStartingBucketId");
+            getBlackboard().signalGate("waiting");
+            try {
+              getBlackboard().waitForGate("done", 1, TimeUnit.MINUTES);
+            } catch (Exception e) {
+              throw new RuntimeException(e);
```

Comment:
Could you add a message here, so that if this timeout is not long enough in the future, we have some understanding of what really went wrong?